### PR TITLE
1350 IHE GW perf 2

### PR DESCRIPTION
**Pull request by leite08**
Ticket: https://github.com/metriport/metriport-internal/issues/1350

### Dependencies

- Upstream: https://github.com/metriport/metriport-internal/pull/1556
- Downstream: none

### Description

- perf adj on IHE GW configs
   - throw on errors
   - only one error processing per SOAP error
   - catch-all
- minor DX updates 

### Testing

- Local
  - [x] PD/DQ/DR work
- Staging
  - [ ] PD/DQ/DR work

### Release Plan

- [ ] Merge upstream
- [ ] Merge this

### CODE REVIEW

**Review by Goncharo — APPROVED** [review]
:shipit:

File: packages/api/src/external/carequality/document/get-non-existent-doc-refs.ts [hunk 1]
```diff
@@ -47,18 +47,12 @@ async function checkDocRefsExistInS3(
...
-        }
-
-        const fileName = createFileName(cxId, patientId, metriportId);
+        const fileName = createFileName(cxId, patientId, doc.metriportId);
 
         const { exists } = await s3Utils.getFileInfoFromS3(fileName, s3BucketName);
 
         successfulDocs.push({
-          docId: metriportId,
+          docId: doc.metriportId,
```
Comment (Goncharo) [inline packages/api/src/external/carequality/document/get-non-existent-doc-refs.ts:55]: why change this? seems like we added this check for a reason?

File: packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts [hunk 2]
```diff
@@ -33,14 +33,14 @@ export async function processOutboundDocumentRetrievalResps({
...
         if (operationOutcome?.issue) {
           issuesWithGateway += operationOutcome.issue.length;
         }
-
-        if (docRetrievalResp.documentReference) {
-          successDocsCount += docRetrievalResp.documentReference.length;
-          await handleDocReferences(docRetrievalResp.documentReference, requestId, patientId, cxId);
+        const docRefs = docRetrievalResp.documentReference;
+        if (docRefs) {
+          const validDocRefs = docRefs.filter(containsMetriportId);
```
Comment (Goncharo) [inline packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts:41]: when wouldn't doc refs contain a metriport ID?

File: packages/api/src/external/carequality/document/get-non-existent-doc-refs.ts (hunk 1 above)
Comment (leite08) [reply packages/api/src/external/carequality/document/get-non-existent-doc-refs.ts:55]: https://metriport.slack.com/archives/C04CXLSAF7V/p1709094435762369?thread_ts=1709075879.242879&cid=C04CXLSAF7V

File: packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts (hunk 2 above)
Comment (leite08) [reply packages/api/src/external/carequality/document/process-outbound-document-retrieval-resps.ts:41]: https://metriport.slack.com/archives/C04CXLSAF7V/p1709094435762369?thread_ts=1709075879.242879&cid=C04CXLSAF7V

**Review by Goncharo — APPROVED** [review]
:shipit:

**Pull request merged**